src/test/java/ro/msg/learning/shop/service/OrderServiceIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+package ro.msg.learning.shop.service;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.DummyData;
+import ro.msg.learning.shop.IntegrationTest;
+import ro.msg.learning.shop.exception.LocationSelectionException;
+import ro.msg.learning.shop.model.Address;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class OrderServiceIntegrationTest extends IntegrationTest {
+
+    @Autowired
+    private OrderService orderService;
+    @Autowired
+    private ProductRepository productRepository;
```

Comment:
Also, the integration tests should test the entire flow from the controllers to the DB. So in the integration tests you should call the endpoints, not the repositories or services.

Reply:
I created integration tests for the OrderController.

Reply:
removed